gitbutler-ui/src/lib/backend/summarizer.ts

```diff
@@ -80,3 +91,45 @@ export class Summarizer {
 		return message;
 	}
 }
+
+interface SummarizerContext {
+	user?: User;
+}
+
+// This optionally returns a summarizer. There are a few conditions for how this may occur
+// Firstly, if the user has opted to use the GB API and isn't logged in, it will return undefined
+// Secondly, if the user has opted to bring their own key but hasn't provided one, it will return undefined
+export async function buildSummarizer(context: SummarizerContext): Promise<Summarizer | undefined> {
+	const gitConfig = new GitConfig();
```

Comment:
There is still something I want to figure out here. If we were testing either of the components where this is used, how would we swap this out for a mock `GitConfig` object? To me it feels like we need an `AiService` created in one of the `layout.ts` files and passed down to `BranchCard` and `CommitDialog`.

Reply:
I'm not a huge fan of drilling the props down. There doesn't seem to be much use of the setContext and getContext API, is there a particular reason for this (and would it be reasonable to provide the GitConfig and AIService this way)?

Also, is AiSomething preferred over AISomething?

Reply:
Yeah I'm not a fan either, and in fact, we're going to move to `setContext` and `getContext`. The only obstacle is that we need to replace the `projectId` constructor params with a store.. or something. I will have a look at that this coming week!

> Also, is AiSomething preferred over AISomething

Yeah, but it's just my personal preference. I don't like having e.g. three capital letters in a row. 😏